src/Symfony/Component/Mapper/AutoMapper.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace Symfony\Component\Mapper;
+
+use Psr\Container\ContainerInterface;
+use SplObjectStorage;
+use Symfony\Component\Mapper\Attributes\Map;
+use Symfony\Component\Mapper\Exception\RuntimeException;
+use Symfony\Component\PropertyAccess\PropertyAccessorInterface;
+
+/**
+ * Object to object mapper.
+ *
+ * @experimental
+ * @author Antoine Bluchet <soyuka@gmail.com>
+ */
+final class AutoMapper implements MapperInterface
+{
+    private SplObjectStorage $objectMap;
+
+    public function __construct(private readonly ?PropertyAccessorInterface $propertyAccessor = null, private readonly ?ContainerInterface $serviceLocator = null)
+    {
+        $this->objectMap = new SplObjectStorage();
```

Comment:
I think this should not be in the constructor, it should be scooped to the map process, otherwise this will be a memory leak when using this class in a worker

To avoid having a context for that, maybe you can create this as a static parameter in the `map` method and unset it when it has been created like : 

```
static $objectMap = null;
$objectMapInitialized = false;

if ($objectMap === null) {
    $objectMap = new SplObjectStorage();
    $objectMapInitialized = true;
}

...

if ($objectMapInitialized) {
   $objectMap = null
}

return $mapped
```

Reply:
You can implement  `Symfony\Contracts\Service\ResetInterface` and clear or reset `$this->objectMap` in the `reset()` method.

Reply:
If we can avoid the `ResetInterface` this it will be better, as it give a new and hidden reponsability to user when using this component in standalone.

Also when i look at the current implementation this storage does not check about target type, so if you do `$foo` to `FooDto` and then `$foo` to `FooEntity` (with the same source object), i believe it will give FooDto in the second mapping (since it will use cache from the first mapping) i'm not sure this can happens in the same mapping scope ? If it does, the target type should be somehow included in the key of this storage.

Reply:
This kind of cache may benefit performances being kept across messages, for this kind of use case I tend to implement LRU or LFU cache strategies instead.

Reply:
done thanks!

Reply:
`ResetInterface` may still be kept, and behavior could be configurable by user.

Reply:
there's no benefits in doing so, this is just to maintain recursive mapping